### DISCUSSION TITLE
In which situations is the "Default Redirect URI" used?

### DISCUSSION THREAD

**Discussion by mrdcb** [zitadel/zitadel#7007] [Q&A]
I stumbled over this when clicking on the login link in the "Password has changed" mail.

I would have expected – since I did set the password redirect link – that I would land in my app after login and not the zitadel console. 

One could argue for or against this, but this makes me wonder: in what situations is the redirect uri used, and when is it not used?

**Comment by hifabienne** [comment]
I assume you mean the redirect url in the auhorization request?
It is used to redirect the user back to the app after authentication. The redirect url in the authorize request has to match with the one configured on the application. this is due to the oidc standard.

When the user clicks the forget password link, and gets an email, a new login ui is opened. In that case we loose the link to the authrequest.

You can configure the default redirect url, which is taken when we do not have an authrequest, in the login behaviour settings.

**Reply by mrdcb** [reply]
Hi thank you for your answer!

I do mean the uri that is configured in the login behaviour settings.

So the expected behaviour would be:
- I set the default redirect uri in my instnace settings to https://example.com/me
- the user recieves a "your password has changed" email and clicks on the login link in there
- the user logs in
- after successfull login the user is redirected to https://example.com/me

That is not the case for me, the user lands on the zitadel ui. I was not sure if this is a bug or not…
I'm happy to create an issue if this would be a better place for this?
(I use a self hosted instance so maybe this is also a configuration issue?)

Thank You

**Reply by hifabienne** [reply]
do you use the default settings on the organization of the user? or can you try to set the url in the organization?

**Reply by mrdcb** [reply]
I did set the same url for the instance and the organisation

**Reply by errnesto** [reply]
So after looking into this a bit more I realized that when accessing `/ui/login/` (or simply to `/`) zitadel will use the "default redirect uri" setup in the settings.

But when accessing `/ui/console` it will redirect to the console after login. (which – of course – makes perfekt sense!)

What is disrupts the flow in our setup is that the `/ui/console` link is used in emails after a password has changed.
I guess what you want to happen here really depends on your setup…

But is it possible to configure that link somewhere?

**Reply by hifabienne** [reply]
Yes this is the problem.
There are two different mails. 
One is the actual password reset request email, which sends the user to a ui to set a new password. I just tested this and the redirect to the default url works.
The second is the password changed email, which notifies the user of a password change. in that case the link points to the console, as this is an information and we want to send the user to his profile. In case that the user didn't change the password himself. 

No at the moment it is not possible to configure the link yourself. In our new api this will be possible, the api is not finished yet but we will provide more and more requests in the future: https://zitadel.com/docs/apis/resources/user_service
